projects/v2/src/app/modules/tree/vega.service.ts

```diff
@@ -71,21 +69,30 @@ export class VegaService {
         this.store.dispatch(new OpenBottomSheet(node));
       }
 
-      const nodeInfo: GaNodeInfo = {
-        oid: node.ontologyId,
-        type: node.type,
-        x: node.x,
-        y: node.y
-      };
-      this.ga.eventEmitter('graph_label_click', GaCategory.GRAPH, 'Clicked a node label', GaAction.CLICK, JSON.stringify(nodeInfo));
+      this.ga.eventEmitter('graph_label_click', GaCategory.GRAPH, 'Clicked a node label', GaAction.CLICK, this.makeNodeInfoString(node));
     });
 
     view.addSignalListener('node__click', (signal: Signal, nodeId: any) => {
-      // TODO get more node information
-      this.ga.eventEmitter('graph_node_click', GaCategory.GRAPH, 'Clicked a node', GaAction.CLICK, nodeId);
+      if (nodeId != null) {
+        console.log("clicked node " + nodeId);
+        const node = this.store.selectSnapshot(TreeState.getBimodal).nodes.find(node => node.id === nodeId);
+        this.ga.eventEmitter('graph_node_select', GaCategory.GRAPH, 'Selected (clicked) a node', GaAction.CLICK, this.makeNodeInfoString(node));
+      } else {
+        this.ga.eventEmitter('graph_node_deselect', GaCategory.GRAPH, 'Deselected a node', GaAction.CLICK);
+      }
     });
   }
 
+  makeNodeInfoString(node: any) {
```

Comment:
Can this be moved to the GA service or something? This tile should technically have only vega and visualization related files.